src/OrchardCore.Modules/OrchardCore.Settings/Services/SetupEventHandler.cs

```diff
@@ -17,23 +19,15 @@ public SetupEventHandler(ISiteService siteService)
         }
 
         public async Task Setup(
-            string siteName,
-            string userName,
-            string userId,
-            string email,
-            string password,
-            string dbProvider,
-            string dbConnectionString,
-            string dbTablePrefix,
-            string siteTimeZone,
+            IDictionary<string, object> properties,
             Action<string, string> reportError
             )
         {
             // Updating site settings
             var siteSettings = await _siteService.LoadSiteSettingsAsync();
-            siteSettings.SiteName = siteName;
-            siteSettings.SuperUser = userId;
-            siteSettings.TimeZoneId = siteTimeZone;
+            siteSettings.SiteName = properties[SetupConstants.SiteName]?.ToString();
```

Comment:
```suggestion
            if (properties.TryGetValue(SetupConstants.SiteName, out string siteName))
            {
                siteSettings.SiteName = siteName;
            }
```

While it's expected that these properties are supplied, this would avoid any key not found errors in the dictionary. 
Where `properties[SetupConstants.SiteName]? would throw a key not found exception